packages/prompts/test/autocomplete.test.ts

```diff
@@ -0,0 +1,124 @@
+import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
+import { autocomplete } from '../src/autocomplete.js';
+import { MockReadable, MockWritable } from './test-utils.js';
+
+describe('autocomplete', () => {
+	let input: MockReadable;
+	let output: MockWritable;
+	const testOptions = [
+		{ value: 'apple', label: 'Apple' },
+		{ value: 'banana', label: 'Banana' },
+		{ value: 'cherry', label: 'Cherry' },
+		{ value: 'grape', label: 'Grape' },
+		{ value: 'orange', label: 'Orange' },
+	];
+
+	beforeEach(() => {
+		input = new MockReadable();
+		output = new MockWritable();
+	});
+
+	afterEach(() => {
+		vi.restoreAllMocks();
+	});
+
+	test('renders initial UI with message and instructions', async () => {
+		const result = autocomplete({
+			message: 'Select a fruit',
+			options: testOptions,
+			input,
+			output,
+		});
+
+		expect(output.buffer).toMatchSnapshot();
+	});
+
+	test('shows placeholder when provided', async () => {
+		const result = autocomplete({
+			message: 'Select a fruit',
+			options: testOptions,
+			placeholder: 'Type to search...',
+			input,
+			output,
+		});
+
+		expect(output.buffer).toMatchSnapshot();
+	});
+
+	test('limits displayed options when maxItems is set', async () => {
+		const result = autocomplete({
+			message: 'Select a fruit',
+			options: testOptions,
+			maxItems: 2,
```

Comment:
we force `maxItems` to be a minimum of 5 internally

so this test should probably generate some options and set the `maxItems` to 6 or something

```ts
const options = [];
for (let i = 0; i < 10; i++) {
  options.push({ value: `option ${i}` });
}
```